### PR TITLE
Frozen and seeded splits

### PR DESCRIPTION
**Pull request by samedii**
- Frozen splits
- Seeded splits

### CODE REVIEW

File: datastream/tools/split_dataframes.py [hunk 1]
```diff
@@ -29,43 +29,50 @@ def split_dataframes(
...
             )
     else:
         split = {
             split_name: list()
             for split_name in proportions.keys()
         }
 
-    split_proportions = tuple(proportions.items())
+    if len(dataframe[key_column].unique()) != len(dataframe):
+        raise KeyError(f'key_column {key_column} contains duplicate values')
```
Comment (Jim-Holmstroem) [inline datastream/tools/split_dataframes.py:46]: https://docs.python.org/3/library/exceptions.html#KeyError
is pretty specific

I think this is more of a ValueError

Comment (samedii) [reply datastream/tools/split_dataframes.py:46]: Maybe I don't care that much but I think this is quite literally a key error? Would you call it a key error if the dataframe was a dictionary instead? I'm open to changing this

Comment (Jim-Holmstroem) [reply datastream/tools/split_dataframes.py:46]: `key is not found in the set of existing keys` is pretty specific (any mapping type), but if the mapping is not on the form expected its something else

File: datastream/tools/split_dataframes.py [hunk 2]
```diff
@@ -29,43 +29,50 @@ def split_dataframes(
...
+                'keys of proportions'
             )
     else:
         split = {
             split_name: list()
             for split_name in proportions.keys()
         }
 
-    split_proportions = tuple(proportions.items())
+    if len(dataframe[key_column].unique()) != len(dataframe):
```
Comment (Jim-Holmstroem) [inline datastream/tools/split_dataframes.py:45]:
```suggestion
    if dataframe[key_column].nunique() != len(dataframe):
```

**Pull request merged**